base/linalg/tridiag.jl

```diff
@@ -498,9 +498,83 @@ function convert{T}(::Type{SymTridiagonal{T}}, M::Tridiagonal)
     end
 end
 
+
 A_mul_B!(C::AbstractVector, A::Tridiagonal, B::AbstractVector) = A_mul_B_td!(C, A, B)
 A_mul_B!(C::AbstractMatrix, A::Tridiagonal, B::AbstractVecOrMat) = A_mul_B_td!(C, A, B)
 A_mul_B!(C::AbstractVecOrMat, A::Tridiagonal, B::AbstractVecOrMat) = A_mul_B_td!(C, A, B)
+A_mul_B!(C::AbstractMatrix, A::Tridiagonal, B::Tridiagonal) = A_mul_B_td!(C, A, B)
+A_mul_B!(C::AbstractMatrix, A::AbstractMatrix, B::Tridiagonal) = A_mul_B_td!(C, A, B)
+
+
+function check_A_mul_B!_sizes(C, A, B)
+    nA, mA = size(A)
+    nB, mB = size(B)
+    nC, mC = size(C)
+    if !(nA == nC)
+        throw(DimensionMismatch("Sizes size(A)=$(size(A)) and size(C) = $(size(C)) must match at first entry."))
+    elseif !(mA == nB)
+        throw(DimensionMismatch("Second entry of size(A)=$(size(A)) and first entry of size(B) = $(size(B)) must match."))
+    elseif !(mB == mC)
+        throw(DimensionMismatch("Sizes size(B)=$(size(B)) and size(C) = $(size(C)) must match at first second entry."))
+    end
+end
+
+function A_mul_B_td!(C::AbstractMatrix, A::Tridiagonal, B::Tridiagonal)
```

Comment:
The first three A_mul_B_td!(C, A, B) were already there, when I started. I guess its to resolve definition ambiguity. For example you have to define
 ```A_mul_B!(C::AbstractVector, A::Tridiagonal, B::AbstractVector)``` 
before 
```A_mul_B!(C::AbstractVecOrMat, A::Tridiagonal, B::AbstractVecOrMat)```
but you only want to write a kernel for the latter. So I don't see how to do this without the `_td` version?